src/VisualStudio/Core/Def/Implementation/Diagnostics/FxCopAnalyzersSuggestedActionCallback.cs

```diff
@@ -0,0 +1,282 @@
+            if (!_infoBarChecked &&
+                !IsVsixInstalled() &&
+                !IsNuGetInstalled() &&
+                IsCandidate())
+            {
+                ShowInfoBarIfNecessary();
+            }
+        }
+
+        private bool IsVsixInstalled()
+        {
+            if (_vsixInstallStatus == FxCopAnalyzersInstallStatus.Unknown)
+            {
+                var vsShell = _serviceProvider.GetService(typeof(SVsShell)) as IVsShell;
+                var hr = vsShell.IsPackageInstalled(FxCopAnalyzersPackageGuid, out var installed);
+                if (ErrorHandler.Failed(hr))
+                {
+                    FatalError.ReportWithoutCrash(Marshal.GetExceptionForHR(hr));
+
+                    // We set installed to ensure we don't go through this again next time a
+                    // suggested action is called, and we don't want to continue if the shell
+                    // is busted.
+                    _vsixInstallStatus = FxCopAnalyzersInstallStatus.Installed;
+                }
+                else
+                {
+                    _vsixInstallStatus = installed != 0 ? FxCopAnalyzersInstallStatus.Installed : FxCopAnalyzersInstallStatus.NotInstalled;
+                    FxCopAnalyzersInstallLogger.LogVsixInstallationStatus(_workspace, _vsixInstallStatus);
+                }
+            }
+
+            return _vsixInstallStatus == FxCopAnalyzersInstallStatus.Installed;
+        }
+
+        private bool IsNuGetInstalled()
+        {
+            if (_nugetInstallStatusForCurrentSolution != FxCopAnalyzersInstallStatus.Installed)
+            {
+                var activeDocumentId = _documentTrackingService.TryGetActiveDocument();
+                if (activeDocumentId == null)
+                {
+                    return false;
+                }
+
+                var document = _workspace.CurrentSolution.GetTextDocument(activeDocumentId);
+                if (document == null)
+                {
+                    return false;
+                }
+
+                foreach (var analyzerReference in document.Project.AnalyzerReferences)
+                {
+                    if (ChildAnalyzerNuGetPackageIds.Contains(analyzerReference.Display))
+                    {
+                        // We set installed to ensure we don't go through this again next time a
+                        // suggested action is called for any document in current solution.
+                        _nugetInstallStatusForCurrentSolution = FxCopAnalyzersInstallStatus.Installed;
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        private bool IsCandidate()
+        {
+            // Candidates fill the following critera:
+            //     1: Are a Dotnet user (as evidenced by the fact that this code is being run)
+            //     2: Have triggered a lightbulb on 3 separate days
```

Comment:
Not sure I understand this. Why do we want to wait for at least 3 days to show info bar?

Reply:
I sync'ed offline with Jinu and we are going to special case code fixes that inherently improve code quality and not wait 3 days if user invokes one of those fixes. Otherwise, we will use the existing logic of waiting 3 days.